### PR TITLE
Powertemp

### PR DESCRIPTION
**Pull request by llGines**
Add get temperature, input power and output power features. Change name Smartlight by Allegro

### CODE REVIEW

File: src/allegro/allegro.py [hunk 1]
```diff
@@ -236,3 +236,28 @@ def interrogate_fiber(
...
         return d
+
+    def get_temp(self) -> float:
+        """Return the temperature in the device."""
+        if not self.connected:
+            raise AllegroConnectionError(CONN_MSG)
+
+        return round(random.uniform(35, 50), 2)
+
+    def get_output_power(self, output_ports: list[int]) -> dict[int, int]:
```
Comment (cpascual) [inline src/allegro/allegro.py:247]: the output should be `dict[int,float]`.

File: src/allegro/allegro.py [hunk 2]
```diff
@@ -236,3 +236,28 @@ def interrogate_fiber(
...
+        """Return the power measured in the given ports."""
+        if not self.connected:
+            raise AllegroConnectionError(CONN_MSG)
+        d = {}
+        for i in output_ports:
+            d[i] = i
+        return d
+
+    def get_input_power(self, input_ports: list[int]) -> dict[int, int]:
```
Comment (cpascual) [inline src/allegro/allegro.py:256]: the output should be dict[int,float].

File: tests/test_api.py [hunk 3]
```diff
@@ -181,27 +181,53 @@ def test_happy_switch(smartlight):
...
+    assert 35 <= t <= 50
+
+
+def test_output_power(allegro):
+    with pytest.raises(AllegroConnectionError):
+        allegro.get_output_power([1, 2, 3])
+    allegro.connect()
+    a = allegro.get_output_power([1, 2, 3])
+    assert isinstance(a, dict)
+    assert set(a.keys()) & set(a.values())
```
Comment (cpascual) [inline tests/test_api.py:224]: I think this test is not doing what you intend

File: tests/test_api.py [hunk 4]
```diff
@@ -181,27 +181,53 @@ def test_happy_switch(smartlight):
...
+
+
+def test_input_power(allegro):
+    with pytest.raises(AllegroConnectionError):
+        allegro.get_input_power([1, 2, 3])
+    allegro.connect()
+    a = allegro.get_input_power([1, 2, 3])
+    assert isinstance(a, dict)
+    assert (b == c * 0.1 for b, c in zip(a.keys(), a.values(), strict=False))
```
Comment (cpascual) [inline tests/test_api.py:233]: I think this test is not doing what you intend

**Review by cpascual — CHANGES_REQUESTED** [review]
Good in general, but there are some changes needed (see open conversations)

File: src/allegro/allegro.py (hunk 1 above)
Comment (llGines) [reply src/allegro/allegro.py]: done

File: src/allegro/allegro.py (hunk 2 above)
Comment (llGines) [reply src/allegro/allegro.py]: done

File: tests/test_api.py (hunk 3 above)
Comment (llGines) [reply tests/test_api.py]: done

**Pull request merged**